components/global/core/io.cellery.observability.siddhi.apps/src/main/siddhi/k8s-component-telemetry-app.siddhi

```diff
@@ -0,0 +1,25 @@
+@App:name("k8s-component-telemetry-app")
+@App:description("This collects cell data and stores them in relevant data sources")
+
+@source(type="k8s-cellery-components",
+        @map(type="keyvalue", fail.on.missing.attribute="false"))
+define stream K8sComponentEventSourceStream(cell String, component string, creationTimestamp long, lastKnownActiveTimestamp long, Ingress_Types string, action string);
+
+@Store(type="rdbms", datasource="CELLERY_OBSERVABILITY_DB")
+@PrimaryKey("cell","component")
```

Comment:
I think the primary key should be cell, component and ingressType. Though we support only one ingressType per component at the moment, we will support multiple ingress types for components in future. Therefore, the primary key should be cell, component, and ingress type.

Reply:
As we are passing the ingress types as comma seperated values to the database, there will not be any issues when storing multiple ingress types for a specific component. Therefore it is not really necessary to make the ingressType as primary key at the moment.